api/api.ts

```diff
@@ -0,0 +1,244 @@
+     *
+     * (Source CMake Documentation cmake-file-api(7))
+     *
+     * \todo clarify need of INTERFACE_LIBRARY type
+     */
+    type: TargetTypeString;
+
+    /** A string specifying the absolute path to the target’s source directory. */
+    sourceDirectory?: string;
+
+    /** Name of the target artifact on disk (library or executable file name). */
+    fullName?: string;
+
+    /** List of absolute paths to a target´s build artifacts. */
+    artifacts?: string[];
+
+    /**
+     * The file groups describe a list of compilation information for artifacts of this target.
+     * The file groups contains source code files that use the same compilation information
+     * and are known by CMake.
+     */
+    fileGroups?: CodeModelFileGroup[];
+
+    /**
+     * Represents the CMAKE_SYSROOT variable
+     */
+    sysroot?: string;
+}
+
+/**
+ * Describes a file group to describe the build settings.
+ */
+export interface CodeModelFileGroup {
+    /** List of source files with the same compilation information */
+    sources: string[];
+
+    /** Specifies the language (C, C++, ...) for the toolchain */
+    language?: string;
+
+    /** Include paths for compilation of a source file */
+    includePath?: {
+        /** include path */
+        path: string;
+    }[];
+
+    /** Compiler flags */
+    compileCommandFragments?: string[];
+
+    /** Defines */
+    defines?: string[];
+
+    /** CMake generated file group */
+    isGenerated: boolean;
+}
+
+/**
+ * Describes cmake project and all its related targets
+ */
+export interface CodeModelProject {
+    /** Name of the project */
+    name: string;
+
+    /** List of targets */
+    targets: CodeModelTarget[];
+
+    /** Location of the Project */
+    sourceDirectory: string;
+
+    hasInstallRule?: boolean; // Exists in ServerCodeModelProject.
+
+}
+
+/**
+ * Describes cmake configuration
+ */
+export interface CodeModelConfiguration {
+    /** List of project() from CMakeLists.txt */
+    projects: CodeModelProject[];
+
+    /** Name of the active configuration in a multi-configuration generator.*/
+    name: string;
+}
+
+export interface CodeModelToolchain {
+    path: string;
+    target?: string;
+}
+
+/** Describes the cmake model */
+export interface CodeModelContent {
+    /** List of configurations provided by the selected generator */
+    configurations: CodeModelConfiguration[];
+
+    toolchains?: Map<string, CodeModelToolchain>;
+}
+
+/**
+ * Helper function to get the CMakeToolsApi from the CMake Tools extension.
+ * @param version The desired API version.
+ * @param exactMatch If true, the version must match exactly. Otherwise, the
+ * function will attempt to return the requested version, but it may not match
+ * exactly.
+ * @returns The API object, or undefined if the API is not available.
+ */
+export async function getCMakeToolsApi(version: Version, exactMatch = true): Promise<CMakeToolsApi | undefined> {
```

Comment:
I prefer `exactMatch` to be false by default.  Otherwise the behavior is different from the cpptools API.

Reply:
Changed default value.